agent/src/iconfucius/cli/chat.py

```
@@ -1322,136 +1328,161 @@ def run_chat(persona_name: str, bot_name: str, verbose: bool = False,
+                # Hot-swap backend when api_type or base_url changed
+                if ai_result is not None:
+                    new_api_type, new_model, new_base_url = ai_result
+                    prev_api_type = persona.ai_api_type
+                    prev_model = persona.ai_model
+                    prev_base_url = persona.ai_base_url
+                    persona.ai_api_type = new_api_type
+                    persona.ai_model = new_model
+                    persona.ai_base_url = new_base_url
+                    try:
+                        new_backend = create_backend(persona)
+                    except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)